src/vs/workbench/browser/actions/layoutActions.ts

```diff
@@ -534,6 +534,99 @@ MenuRegistry.appendMenuItem(MenuId.MenubarAppearanceMenu, {
 	order: 10
 });
 
+// --- Show Editor Actions in Title Bar
+
+export class EditorActionsTitleBarAction extends Action2 {
+
+	static readonly ID = 'workbench.action.editorActionsTitleBar';
+
+	constructor() {
+		super({
+			id: EditorActionsTitleBarAction.ID,
+			title: {
+				value: localize('moveEditorActionsToTitleBar', "Move Editor Actions to Title Bar"),
+				original: 'Move Editor Actions to Title Bar'
+			},
+			category: Categories.View,
+			precondition: ContextKeyExpr.and(
+				ContextKeyExpr.equals('config.workbench.editor.editorActionsLocation', 'titleBar').negate(),
+				ContextKeyExpr.equals('config.window.style', 'native').negate(),
+			),
+			f1: true
+		});
+	}
+
+	run(accessor: ServicesAccessor): Promise<void> {
+		const configurationService = accessor.get(IConfigurationService);
+		return configurationService.updateValue('workbench.editor.editorActionsLocation', 'titleBar');
+	}
+}
+registerAction2(EditorActionsTitleBarAction);
+
+// --- Editor Actions Default Position
+
+export class EditorActionsDefaultAction extends Action2 {
+
+	static readonly ID = 'workbench.action.editorActionsDefault';
+
+	constructor() {
+		super({
+			id: EditorActionsDefaultAction.ID,
+			title: {
+				value: localize('moveEditorActionsToTabBar', "Move Editor Actions to Tab Bar"),
+				original: 'Move Editor Actions to Tab Bar'
+			},
+			category: Categories.View,
+			precondition: ContextKeyExpr.and(
+				ContextKeyExpr.equals('config.workbench.editor.editorActionsLocation', 'default').negate(),
+				ContextKeyExpr.equals('config.workbench.editor.showTabs', 'none').negate(),
+			),
+			f1: true
+		});
+	}
+
+	run(accessor: ServicesAccessor): Promise<void> {
+		const configurationService = accessor.get(IConfigurationService);
+		return configurationService.updateValue('workbench.editor.editorActionsLocation', 'default');
+	}
+}
+registerAction2(EditorActionsDefaultAction);
+
+// --- Hide Editor Actions
+
+export class HideEditorActionsAction extends Action2 {
+
+	static readonly ID = 'workbench.action.editorActionsHidden';
+
+	constructor() {
+		super({
+			id: HideEditorActionsAction.ID,
+			title: {
+				value: localize('hideEditorActons', "Hide Editor Actions"),
```

Comment:
Should we have a related "Show Editor Actions"? I know we have the "Move..." commands but maybe people would expect to search for "Show...".

Reply:
I agree, that makes sense